### PR TITLE
[sql lab] WIP. Autosync queries periodically

### PR DESCRIPTION
**Pull request by bkyryliuk**
Implements https://github.com/airbnb/caravel/issues/1016

### CODE REVIEW

File: caravel/assets/javascripts/SqlLab/components/QueryHistory.jsx [hunk 1]
```diff
@@ -18,7 +18,8 @@ const QueryHistory = (props) => {
   if (queriesArray.length > 0) {
     return (
       <QueryTable
-        columns={['state', 'started', 'duration', 'rows', 'sql', 'actions']}
+        columns={['state', 'started', 'duration', 'progress', 'rows', 'sql',
```
Comment (mistercrunch) [inline caravel/assets/javascripts/SqlLab/components/QueryHistory.jsx]: Isn't there room on that line? linter says 100 I think

File: caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 2]
```diff
@@ -56,36 +56,63 @@ class SqlEditor extends React.Component {
...
-      data,
+      url: createQueryUrl,
+      data: createQueryRequest,
       success(results) {
-        if (runAsync) {
-          // TODO nothing?
-        } else {
-          that.props.actions.querySuccess(query, results);
-        }
+        query = Object.assign({}, query, results['query'])
```
Comment (mistercrunch) [inline caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx]: Let's let the reducer manage the objectids

File: caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 3]
```diff
@@ -56,36 +56,63 @@ class SqlEditor extends React.Component {
...
+
+        // Execute the Query
+        that.props.actions.startQuery(query);
+
+        const sqlJsonUrl = '/caravel/sql_json/';
```
Comment (mistercrunch) [inline caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx]: No nested callbacks!

File: caravel/sql_lab_utils.py [hunk 4]
```diff
@@ -14,7 +14,7 @@ def create_scoped_session():
     engine = create_engine(
         app.config.get('SQLALCHEMY_DATABASE_URI'), convert_unicode=True)
     return scoped_session(sessionmaker(
-        autocommit=True, autoflush=False, bind=engine))
+        autocommit=False, autoflush=False, bind=engine))
```
Comment (mistercrunch) [inline caravel/sql_lab_utils.py]: mmmh, not sure why this is necessary or the implications of that

File: caravel/views.py [hunk 5]
```diff
@@ -1444,45 +1444,23 @@ def theme(self):
         return self.render_template('caravel/theme.html')
 
     @has_access
-    @expose("/sql_json/", methods=['POST', 'GET'])
+    @expose("/create_query/", methods=['POST', 'GET'])
```
Comment (mistercrunch) [inline caravel/views.py]: I don't think this endpoint is needed

File: caravel/assets/javascripts/SqlLab/components/QueryHistory.jsx (hunk 1 above)
Comment (bkyryliuk) [inline caravel/assets/javascripts/SqlLab/components/QueryHistory.jsx]: good point, I used python config for it.

File: caravel/assets/javascripts/SqlLab/reducers.js [hunk 6]
```diff
@@ -191,8 +190,10 @@ export const sqlLabReducer = function (state, action) {
       return removeFromArr(state, 'alerts', action.alert);
     },
     [actions.REFRESH_QUERIES]() {
-      queries = action.queries
-      return alterInArr(state, 'queries', queries, { state: 'stopped' });
+      const newQueries = Object.assign({}, state['queries'],
+          action.alteredQueries);
+      console.log(newQueries)
+      return Object.assign({}, state, { ['queries']: newQueries });
```
Comment (mistercrunch) [inline caravel/assets/javascripts/SqlLab/reducers.js]: ES6 is cool but no need to use the feature here :),  `{ queries: newQueries }` should do the exact same thing

File: caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 2 above)
Comment (bkyryliuk) [inline caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx]: I'll init query right here from the server - const query = Object.assign({}, results['query'])
removed other assignments as they are not needed.
I am not sure how to let the reducer manage the objectids.
Could you please explain me in a bit more details ?

File: caravel/sql_lab_utils.py (hunk 4 above)
Comment (bkyryliuk) [inline caravel/sql_lab_utils.py]: will put it back.

File: caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 3 above)
Comment (bkyryliuk) [inline caravel/assets/javascripts/SqlLab/components/SqlEditor.jsx]: @mistercrunch 
Do you  have idea how to retrieve query_id from the server without nested callback ?

File: caravel/sql_lab_utils.py (hunk 4 above)
Comment (bkyryliuk) [inline caravel/sql_lab_utils.py]: server was complaining about the transactions and wasn't updating the DB.
I've turned it off and commit manually in the sql_lab.py

**Comment by bkyryliuk** [comment]
@mistercrunch - I've implemented merging logic.
The fact that I change the query id in the object broke the results display,
Do you know how could I fix that?

**Comment by mistercrunch** [comment]
@bkyryliuk let's sit down together and push this through today

**Pull request closed without merging**